lib/classes/Variables.js

```diff
@@ -683,24 +683,20 @@ class Variables {
       // eslint-disable-next-line global-require, import/no-dynamic-require
       const jsFile = require(referencedFileFullPath);
       const variableArray = variableString.split(':');
-      let returnValueFunction;
+      let returnValue;
       if (variableArray[1]) {
         let jsModule = variableArray[1];
         jsModule = jsModule.split('.')[0];
-        returnValueFunction = jsFile[jsModule];
+        returnValue = jsFile[jsModule];
       } else {
-        returnValueFunction = jsFile;
+        returnValue = jsFile;
```

Comment:
I covered this in my comment on 1914 but you may not have an object or even a concrete value.  What if `jsFile === Error('')`?

Reply:
Totally possible, technically an `Error` would be an object, though perhaps not one that we'd want to insert the values of.  Same issue could happen though if a function returned an `Error` too.  So what should be the handling of that?  We can't prevent the users from returning whatever they want, so I'm not sure whether we want to continue to expand the list of exceptions.

A similar though different use case is what if the file being loaded _throws_ and error.  I think currently we just let that bubble up normally, though it's not wrapped in a Serverless error class at the moment.

Reply:
I think you're calling me out for being a bit too pedantic and I agree.  Reasonableness and dev awareness was assumed before and I shouldn't ask for elevation above that previous bar.  The previous paradigm was to check for the one type of valid return (i.e. a function) but not to vet the result of the function's execution.  Perhaps some positive case checking makes sense but since a non-function return is the equivalent of the function return value, I feel like I have little standing to request more.  Mariusz may have a stronger opinion.

For the record and what it's worth, I don't have merge rights and can't give the green light so I'm just attempting to be helpful.  Despite that, sorry for the slow responses.

Reply:
@erikerikson great thanks for that input. Indeed there can be anything exported by a module (even `undefined`).

But as long as we do not validate the value that is returned by eventual function resolver, for consistency we also should not validate a directly provided value.

I think real concern comes from a design proposal: where we treat a given value as resolved as long as it's not a _function_ and otherwise we treat a _function_ as value resolver.

I have mixed feelings on that, it definitely can turn to be confusing, e.g. in current form we accept also a function as final resolved value (it can be returned by function resolver)

It'll work better if we restrict and validate the final `valueToPopulate`. Ideally we should just support JSON values, and simplified validation of that would be to reject all values for which `typeof` returns `'undefined`, `'symbol'` or `'function'`.

Reply:
@erikerikson, @medikoo thank you both for the feedback!  Sorry it's taken a bit to respond, I'll try to stay on top of this PR a bit better 😅 

@medikoo I think we're in agreement here - that currently we don't really check the type of `valueToPopulateResolved`, regardless of how that value is provided (function invocation or otherwise).  Focusing on `valueToPopulateResolved` and evaluating whether we've landed on a valid result type makes a lot of sense to me.

If I'm not mistaken, it looks like we already handle this for `undefined` values: https://github.com/serverless/serverless/blob/master/lib/classes/Variables.js#L728.  I've updated this PR to also throw an error if the resolved value is a `symbol` or `function`.